src/components/NavBar.js

```diff
@@ -1,31 +1,63 @@
-import React from "react";
-import {BrowserRouter as Router, Link} from "react-router-dom";
+import React, {useState} from "react";
+import {Link} from "react-router-dom";
 import { faCaretDown } from "@fortawesome/free-solid-svg-icons";
+import { faQuestionCircle } from "@fortawesome/free-solid-svg-icons";
+import { faHammer } from "@fortawesome/free-solid-svg-icons";
+import { faTools } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
 import './NavBar.css';
 import OfekLogo from '../assets/images/OfekLogo.png';
 
-const NavBar = () => {
+const NavBar = props => {
+    
+
+    const toggleSideBar = () => {
+        props.MinMaxMain();
+        if(props.mini) document.getElementById("sidebar").style.width = "250px";
+        else document.getElementById("sidebar").style.width = "85px";
+    };
+
   return (
-    <Router>
-        <nav id="nav" className="nav" role="navigation">
-            <ul className="nav-links">
-                <li>
-                    <Link to="/" className="navItem"><img className="ofeklogo" src={OfekLogo} alt="ofek unit logo" /></Link>
-                </li>
-                <li>
-                    <Link to="/about" className="navItem">About us</Link>
-                </li>
-                <li>
-                    <Link to="/pybuild" className="navItem">PyBuild</Link>
-                </li>
-                <li>
-                    <Link to="/tools" className="navItem"><FontAwesomeIcon icon={faCaretDown} style={{marginRight: "10px"}} />Tools</Link>
-                </li>
-            </ul>
+      <div>
+        <nav id="sidebar" className="sidebar" role="navigation"
+             onMouseOver={() => toggleSideBar()} onMouseOut={() => toggleSideBar()}>
```

Comment:
toggle is a perfect timing to useState with boolean i think

Reply:
Happens in parent component